sdk/java/pulumi/src/main/java/com/pulumi/experimental/automation/UpdateOptions.java

```diff
@@ -0,0 +1,432 @@
+     * also supported.
+     *
+     * @return the list of target resource URNs
+     */
+    public List<String> getTargets() {
+        return targets;
+    }
+
+    /**
+     * A list of paths to policy packs to run during the operation.
+     *
+     * @return the list of paths to policy packs
+     */
+    public List<String> getPolicyPacks() {
+        return policyPacks;
+    }
+
+    /**
+     * A list of paths to policy pack JSON configuration files to use during the
+     * operation.
+     *
+     * @return the list of paths to policy pack JSON configuration files
+     */
+    public List<String> getPolicyPackConfigs() {
+        return policyPackConfigs;
+    }
+
+    /**
+     * Optional callback which is invoked whenever StandardOutput is written into.
+     *
+     * @return the callback
+     */
+    @Nullable
+    public Consumer<String> getOnStandardOutput() {
+        return onStandardOutput;
+    }
+
+    /**
+     * Optional callback which is invoked whenever StandardError is written into.
+     *
+     * @return the callback
+     */
+    @Nullable
+    public Consumer<String> getOnStandardError() {
+        return onStandardError;
+    }
+
+    /**
+     * Optional callback which is invoked with the engine events.
+     *
+     * @return the callback
+     */
+    @Nullable
+    public Consumer<EngineEvent> getOnEvent() {
+        return onEvent;
+    }
+
+    /**
+     * Colorize output. Choices are: always, never, raw, auto (default "auto")
+     *
+     * @return the colorize output setting
+     */
+    @Nullable
+    public String getColor() {
+        return color;
+    }
+
+    /**
+     * Flow log settings to child processes (like plugins)
+     *
+     * @return the flow log settings
+     */
+    public boolean isLogFlow() {
+        return logFlow;
+    }
+
+    /**
+     * Enable verbose logging (e.g., v=3); anything &gt;3 is very verbose
+     *
+     * @return the log verbosity
+     */
+    @Nullable
+    public Integer getLogVerbosity() {
+        return logVerbosity;
+    }
+
+    /**
+     * Log to stderr instead of to files
+     *
+     * @return the log to stderr setting
+     */
+    public boolean isLogToStdErr() {
+        return logToStdErr;
+    }
+
+    /**
+     * Emit tracing to the specified endpoint. Use the file: scheme to write tracing
+     * data to a local file
+     *
+     * @return the tracing endpoint
+     */
+    @Nullable
+    public String getTracing() {
+        return tracing;
+    }
+
+    /**
+     * Print detailed debugging output during resource operations
+     *
+     * @return the debug setting
+     */
+    public boolean isDebug() {
+        return debug;
+    }
+
+    /**
+     * Format standard output as JSON not text.
+     *
+     * @return the JSON output setting
+     */
+    public boolean isJson() {
+        return json;
+    }
+
+    /**
+     * Base builder for all update options.
+     */
+    public static abstract class Builder<B extends Builder<B>> {
```

Comment:
We don't have to do this pre-go-live but I think we should discuss not doing this when this out the door. I don't think the amount of code saved by inheriting e.g. `DestroyOptions` from `UpdateOptions` and the clever `Builder` inheritance tricks is _super_ worth it. I appreciate that we already do this in e.g. `ResourceOptions`, though it's a bit messy there and has made certain kinds of changes fiddly in the past.